### PR TITLE
add: routes

### PR DESCRIPTION
**Pull request by saurabhchirde**
Added React Routes and fetching products from the database.

Files to review:

1. App.js
https://github.com/saurabhchirde/pebble-cart-ecom-react/compare/react-routes?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67
2. Products Provider - to Fetch product list:
https://github.com/saurabhchirde/pebble-cart-ecom-react/compare/react-routes?expand=1#diff-890084c09bff78e285af7fbb8e8cbd7444b97a7ec83de7bd72cfca2fb1707be5

### CODE REVIEW

File: src/Context/ProductsProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const ProductProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    const getproducts = async () => {
+      try {
+        const response = await axios.get("/api/products");
+        setProducts(response.data.products);
+      } catch (error) {
+        console.log(error);
```
Comment (narharikale) [inline src/Context/ProductsProvider.jsx:46]: just a suggestion , we can give some message with error , will be helpful while debugging

**Review by narharikale — APPROVED** [review]
code is clean ✅

File: src/App.js [hunk 2]
```diff
@@ -9,11 +9,14 @@ import "./App.css";
 import Footer from "./Components/UI/Footer/Footer";
 import BodyWrapper from "./Components/UI/Wrapper/BodyWrapper";
 import ProductListingPage from "./Pages/ProductListing/ProductListingPage";
-import { usePageProvider } from "./Context/PageContext";
+import WishlistPage from "./Pages/Wishlist/WishlistPage";
+import CartPage from "./Pages/Cart/CartPage";
+import CheckoutPage from "./Pages/Checkout/CheckoutPage";
```
Comment (manojsarna) [inline src/App.js:14]: Suggestion : can use index.js to export directly in one file.

File: src/Context/ProductsProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        productDispatch({
+          type: "getproducts",
+          payload: response.data.products,
+        });
+        sessionStorage.setItem(
+          "productList",
+          JSON.stringify(productState.products)
+        );
+      } catch (error) {
+        console.log(error);
```
Comment (manojsarna) [inline src/Context/ProductsProvider.jsx:46]: Suggestion : to elaborate on the error part use error.msg or error.response.data

File: src/Context/ProductsProvider.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  useEffect(() => {
+    const getbrands = async () => {
+      try {
+        const response = await axios.get("/api/brands");
+        productDispatch({ type: "getbrands", payload: response.data.brands });
+        sessionStorage.setItem("brands", JSON.stringify(productState.brands));
+      } catch (error) {
+        console.log(error);
+      }
```
Comment (manojsarna) [inline src/Context/ProductsProvider.jsx:79]: Persisting data is good!!

**Review by manojsarna — APPROVED** [review]
Overall it looks good!! 

Do check the suggestions!!

File: src/App.js (hunk 2 above)
Comment (saurabhchirde) [reply src/App.js:14]: Will do it after finishing the project.

File: src/Context/ProductsProvider.jsx (hunk 3 above)
Comment (saurabhchirde) [reply src/Context/ProductsProvider.jsx:46]: Noted.

**Pull request merged**